File: Back/logistica-clientes/src/main/java/com/logistica/app/clientes/service/ClienteService.java
package com.logistica.app.clientes.service;
import com.logistica.commons.service.CommonService;
import com.logistica.app.clientes.entity.Cliente;
import java.math.BigDecimal;
import java.util.List;
public interface ClienteService extends CommonService<Cliente> {
    List<Cliente> buscar(String texto);
    List<Cliente> findByTipo(Cliente.TipoCliente tipo);
    Cliente actualizarAcumulado(Long id, BigDecimal monto);
    Cliente evaluarYActualizarTipo(Long id);
}

File: Front/src/app/services/cliente.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { BASE_ENDPOINT } from "../config/app";
import { Cliente } from "../models/cliente";
import { CommonService } from "./common.service";
@Injectable({ providedIn: "root" })
export class ClienteService extends CommonService<Cliente> {
  protected override baseEndPoint = `${BASE_ENDPOINT}/clientes`;
  constructor(http: HttpClient) { super(http); }
  buscar(texto: string): Observable<Cliente[]> { return this.http.get<Cliente[]>(`${this.baseEndPoint}/buscar/${texto}`); }
  porTipo(tipo: string): Observable<Cliente[]> { return this.http.get<Cliente[]>(`${this.baseEndPoint}/tipo/${tipo}`); }
}